tests/phpunit/integration/Modules/Analytics_4/Synchronize_PropertyTest.php

```diff
@@ -0,0 +1,214 @@
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Modules\Analytics_4;
+use Google\Site_Kit\Modules\Analytics_4\Settings as Analytics_4_Settings;
+use Google\Site_Kit\Modules\Analytics\Settings;
+use Google\Site_Kit\Modules\Analytics_4\Synchronize_Property;
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Tests\FakeHttp;
+use Google\Site_Kit_Dependencies\GuzzleHttp\Psr7\Request;
+use Google\Site_Kit_Dependencies\GuzzleHttp\Psr7\Response;
+
+/**
+ * @group Modules
+ * @group Analytics_4
+ */
+class Synchronize_PropertyTest extends TestCase {
+
+	/**
+	 * @var Synchronize_Property
+	 */
+	protected $synchronize_property;
+
+	/**
+	 * @var Analytics_4
+	 */
+	protected $analytics_4;
+
+	/**
+	 * @var User_Options
+	 */
+	protected $user_options;
+
+	/**
+	 * @var Options
+	 */
+	protected $options;
+
+	/**
+	 * @var Authentication
+	 */
+	protected $authentication;
+
+	/**
+	 * @var \WP_User
+	 */
+	protected $administrator;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->administrator = self::factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+
+		$context              = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options        = new Options( $context );
+		$this->user_options   = new User_Options( $context, $this->administrator->ID );
+		$this->authentication = new Authentication( $context, $this->options, $this->user_options );
+
+		// Fake a valid authentication token on the client.
+		$this->authentication->get_oauth_client()->set_token( array( 'access_token' => 'valid-auth-token' ) );
+
+		$this->analytics_4 = new Analytics_4(
+			$context,
+			$this->options,
+			$this->user_options,
+			$this->authentication
+		);
+
+		$this->set_analytics_settings();
+
+		$this->synchronize_property = new Synchronize_Property(
+			$this->analytics_4,
+			$this->user_options,
+			$this->authentication
+		);
+	}
+
+	public function set_analytics_settings() {
+		// Ensure Anlytics 4 module is connected
+		$this->options->set(
+			Analytics_4_Settings::OPTION,
+			array(
+				'accountID'       => '12345678',
+				'propertyID'      => '987654321',
+				'webDataStreamID' => '1234567890',
+				'measurementID'   => 'A1B2C3D4E5',
+			)
+		);
+
+		// Set owner ID, using base analytics settings.
+		$this->options->set(
+			Settings::OPTION,
+			array(
+				'ownerID' => $this->administrator->ID,
+			)
+		);
+	}
+
+	public function fake_analytics_4_property_object( $property_id, $create_time ) {
+		FakeHttp::fake_google_http_handler(
+			$this->analytics_4->get_client(),
+			function ( Request $request ) use ( $property_id, $create_time ) {
+				$url = parse_url( $request->getUri() );
+
+				if ( "/v1beta/properties/{$property_id}" === $url['path'] ) {
+					return new Response(
+						200,
+						array(),
+						json_encode(
+							array(
+								'name'       => "properties/{$property_id}",
+								'createTime' => $create_time,
+							)
```

Comment:
I like to use the Google Model instances for this, that way we're not testing against our own mock structure.